internal/daemon/api_services.go

```diff
@@ -22,10 +22,36 @@ import (
 
 	"github.com/canonical/pebble/internal/overlord/servstate"
 	"github.com/canonical/pebble/internal/overlord/state"
+	"github.com/canonical/pebble/internal/strutil"
 )
 
+type serviceInfo struct {
+	Name    string `json:"name"`
+	Default string `json:"default"`
+	Status  string `json:"status"`
+	Message string `json:"message,omitempty"`
+}
+
 func v1GetServices(c *Command, r *http.Request, _ *userState) Response {
-	return statusBadRequest("not implemented")
+	names := strutil.CommaSeparatedList(r.URL.Query().Get("names"))
+
+	servmgr := c.d.overlord.ServiceManager()
+	services, err := servmgr.Services(names)
+	if err != nil {
+		return statusInternalError(err.Error())
+	}
+
+	var infos []serviceInfo
```

Comment:
probably could be `infos := make([]serviceInfo, 0, len(services))` but that's a bit of premature optimization.

Reply:
Agreed it's probably premature optimization, but doesn't hurt - done.